### ISSUE TITLE
List manipulation error in saml_credentials_provider.py

### ISSUE THREAD

**Issue by kylemcleland-fanduel** [aws/amazon-redshift-python-driver#56]
# Solution 
In the file `/redshift_connector/plugin/saml_credentials_provider.py`, the following line is causing an error:
```
  for attr in attrs:
      name: str = attr.attrs["Name"]
      values: typing.Any = attr.findAll("{}AttributeValue".format(SAML_RESP_NAMESPACES[namespace_used_idx]))
      if len(values) == 0:
          # Ignore empty-valued attributes.
          continue
      value: str = values[0].contents[0]
```
In one loop, I printed out both `values` and `values[0].content` and returned the following:

>values = [<saml2:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string"/>]
>values[0].content = []

Because values[0].content is an empty list, there is no such thing as values[0].contents[0], therefore assinging that to the string is failing. The check above looks like it's meant to catch this case, but it's only catching when the parent is empty, not if the `.contents` method is empty.

What has worked for me locally is changing `if len(values) == 0:` to `if not values[0].contents:` Because there can be a response for values, but the contents may be empty (NoneType). 

## Problem description
After forking the code and implementing the resolution stated here, https://github.com/aws/amazon-redshift-python-driver/issues/55, the following issue now occurrs. 

Running the following code:

```
import redshift_connector

gdp_conn = redshift_connector.connect(
    iam=True,
    database="mydb",
    cluster_identifier="mycluster",
    credentials_provider='OktaCredentialsProvider',
    user="myuser@domain.com",
    password="mypassword",
    idp_host="myidp",
    app_id="someverylong/string,
    region="us-east-1",
    role_arn="arn:aws:iam::{myaccountid}:role/{my_role_to_assume}",
)
```
the following error occurrs:
Traceback (most recent call last):
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/plugin/saml_credentials_provider.py", line 60, in get_credentials
    self.refresh()
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/plugin/saml_credentials_provider.py", line 164, in refresh
    raise e
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/plugin/saml_credentials_provider.py", line 135, in refresh
    credentials.set_metadata(self.read_metadata(doc))
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/plugin/saml_credentials_provider.py", line 218, in read_metadata
    value: str = values[0].contents[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "redshift_test.py", line 5, in <module>
    gdp_conn = redshift_connector.connect(
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/__init__.py", line 327, in connect
    IamHelper.set_iam_properties(info)
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/iam_helper.py", line 196, in set_iam_properties
    IamHelper.set_iam_credentials(info)
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/iam_helper.py", line 283, in set_iam_credentials
    credentials: CredentialsHolder = provider.get_credentials()
  File "/Users/kylemcleland/.pyenv/versions/3.8.10/lib/python3.8/site-packages/redshift_connector/plugin/saml_credentials_provider.py", line 63, in get_credentials
    raise InterfaceError(e)
redshift_connector.error.InterfaceError: list index out of range
```

**Comment by kylemcleland-fanduel** [comment]
@Brooke-white 

I've identified two issues #55  and #56 (this one), and have come up with a solution that works on my end. I was going to try to create a PR and suggest some changes but I noticed that I don't have the permissions to create a pull request to this repo.

How do you suggest I continue? In the meantime, I am going to use my own fork, but I think my fixes could help others trying to use the same login method.

**Comment by Brooke-white** [comment]
Hey @kylemcleland-fanduel ,

Thank you for diving deep into the code base and debug logs to determine a cause for this issue! We'd love for you to open a PR so you can receive recognition for the work you've done for  #55 and #56.

From our end I am not seeing any restrictions that would prevent you from opening a PR. Have you tried opening a PR from a branch on your fork which contains the fix? Let me know if you're still seeing issue with this and I can take another look.

For the fix for #56, my one suggestion would be to use `if len(values) == 0 or not values[0].contents`,  to ensure we don't proceed in trying to read the SAML attributes in either case (i.e. if there are no attributes or if an attribute is mapped to an empty string)

For the fix for #55, I'd like if we can make this parsing more flexible (i.e. able to handle 0 or more spaces between the ARNs). I think we could do that by doing something like below, where we strip any whitespace from each ARN.

```python
                for value in values:
                    arns = value.contents[0].split(",")
                    role: str = ""
                    provider: str = ""
                    for arn in arns:
                        arn = arn.strip() # remove any leading or trailing whitespace
                        ...
```

This should help prevent any extraneous whitespace from causing a headache :)

Thank you again for doing all the work in identifying this issue, we really appreciate it :)

**Comment by kylemcleland-fanduel** [comment]
Thanks @Brooke-white, I'd be happy to make these changes. I realize that my changes were very specific to my own use case and making them a little more general is a great way to go about it, so thanks for the suggestions on ways to do that.

Unfortunately it does seem I am restricted from writing to this repo. Trying to push a new branch with changes from the command line, I receive the following:

```
remote: Permission to aws/amazon-redshift-python-driver.git denied to kylemcleland-fanduel.
fatal: unable to access 'https://github.com/aws/amazon-redshift-python-driver.git/': The requested URL returned error: 403
```

And just confirming that it's not an issue with my configuration, I tried to just edit a random file through the github UI and was presented with this confirming I don't have access:

![Screen Shot 2021-09-15 at 1 07 34 PM](https://user-images.githubusercontent.com/51718761/133502325-fbe85d92-3f1e-4002-9c83-f570ec359e3e.png)

**Comment by kylemcleland-fanduel** [comment]
Thanks @Brooke-white, you're right I do have permissions. I've done some testing with the changes as how you've proposed them and opened a PR! I'm happy to read any feedback in that PR #57

**Comment by Brooke-white** [comment]
this fix has been included in v2.0.888, thanks Kyle!
